lib/nd_list.dart

```diff
@@ -705,6 +725,36 @@ extension NumNDList on NDList {
   }
 }
 
+class RollingResult<X> {
+  List<NDIndexResult<X>> slices;
+  NDList<X> baseArray;
+
+  RollingResult._(this.slices, this.baseArray);
+
+  factory RollingResult(NDList<X> baseArray, int windowSize,
+      {int step = 1, int axis = 0}) {
+    // allow for negative axis to mean "from end"
+    // eg -1 means "last axis"
+    axis = axis % baseArray.nDims;
+    final slices = [
+      for (int i = windowSize - 1; i < baseArray.shape[axis]; i = i + step)
+        NDList._stringIndex(NDIndexResult.from(baseArray),
```

Comment:
Note: this line requires this class to be in the same file as NDList, so that we can call the private method.